### ISSUE TITLE
random_walker raises ValueError in v0.14.1, while works in v0.13.1  

### ISSUE THREAD

**Issue by winston-li** [scikit-image/scikit-image#3473]
## Description
If there are M objects in an image to segment, and N markers are provided where M > N.
If the N markers are exactly the same as N objects of those M objects in the image, random_walker would be OK in v0.13.1, while failed in v0.14.  

Traceback
```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-4-814636d05dfa> in <module>()
     14 
     15 markers[image == 0] = -1
---> 16 output_labels = random_walker(image, markers)

/opt/conda/lib/python3.6/site-packages/skimage/segmentation/random_walker_segmentation.py in random_walker(data, labels, beta, mode, tol, copy, multichannel, return_full_prob, spacing)
    456     if np.any(labels < 0):
    457         lap_sparse = _build_laplacian(data, spacing, mask=labels >= 0,
--> 458                                       beta=beta, multichannel=multichannel)
    459     else:
    460         lap_sparse = _build_laplacian(data, spacing, beta=beta,

/opt/conda/lib/python3.6/site-packages/skimage/segmentation/random_walker_segmentation.py in _build_laplacian(data, spacing, mask, beta, multichannel)
    190                                   multichannel=multichannel)
    191     if mask is not None:
--> 192         edges, weights = _mask_edges_weights(edges, weights, mask)
    193     lap = _make_laplacian_sparse(edges, weights)
    194     del edges, weights

/opt/conda/lib/python3.6/site-packages/skimage/segmentation/random_walker_segmentation.py in _mask_edges_weights(edges, weights, mask)
    175     ind_mask = np.logical_and(mask0, mask1)
    176     edges, weights = edges[:, ind_mask], weights[ind_mask]
--> 177     max_node_index = edges.max()
    178     # Reassign edges labels to 0, 1, ... edges_number - 1
    179     order = np.searchsorted(np.unique(edges.ravel()),

/opt/conda/lib/python3.6/site-packages/numpy/core/_methods.py in _amax(a, axis, out, keepdims)
     24 # small reductions
     25 def _amax(a, axis=None, out=None, keepdims=False):
---> 26     return umr_maximum(a, axis, None, out, keepdims)
     27 
     28 def _amin(a, axis=None, out=None, keepdims=False):

ValueError: zero-size array to reduction operation maximum which has no identity
```

## Example code to reproduce
```
import numpy as np
import itertools
from skimage.segmentation import random_walker

image = np.full((10, 10), False)
object_A = [(6,7), (6,8), (7,7), (7,8)]
object_B = [(3,1), (4,1), (2,2), (3,2), (4,2), (2,3), (3,3)]
for x, y in itertools.chain(object_A, object_B):
    image[y][x] = True

markers = np.zeros((10, 10), dtype=np.int8)
for x, y in object_B:
    markers[y][x] = 1

markers[image == 0] = -1
output_labels = random_walker(image, markers)
```

**Comment by jni** [comment]
Weird! I don't recall changing `random_walker` code recently, but that means very little in itself! =P

Thanks for the report, @winston-li, we'll investigate and report back here!

**Comment by emmanuelle** [comment]
@jni the change might have been introduced in https://github.com/scikit-image/scikit-image/commit/25ec5a4c2e575ff05853613f3fd0372d87edbe39#diff-f5e00dc483bcaeb6c1b8278dbce09c68 
I will investigate it.

**Comment by emmanuelle** [comment]
@winston-li in the example shown, the random walker algorithm cannot do anything, because the only undetermined pixels (0-valued pixels of the mask) are not connected to any seed pixel (values of the mask > 0), they are only connected to -1 values (corresponding to masked pixels, not taken into account). So, although the error message is cryptic and should be improved, I would not have expected anything meaningful from the case you provided. Did you expect something else?

**Comment by winston-li** [comment]
@emmanuelle  I ran into it while doing post-processing of predictions (objects and markers) of neural networks. I'd expect it worked like v0.13.1 in this case, which random_walker returns the pixels labeled the same as provided markers. Definitely hope not get error/exception, as it indeed happens in some real cases, where some objects to be segmented are exactly the same as all provided markers per the inferences of neural networks. Thanks a lot.

**Comment by emmanuelle** [comment]
@winston-li ok, but then the problem here is that 0 corresponds to unmarked pixels, not to a label, and that these zero-valued pixels are not connected to any markers so that it's impossible to attribute them a label. Does that make sense?

**Comment by winston-li** [comment]
@emmanuelle Yes. Those zero-valued pixels are not connected to any markers, so it's reasonable not to attribute them a label. In v0.13.1, the output_labels in the example code is 
```
    [[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1]
     [-1 -1 -1  1  1 -1 -1 -1 -1 -1]
     [-1 -1  1  1  1 -1 -1 -1 -1 -1]
     [-1 -1  1  1 -1 -1 -1 -1 -1 -1]
     [-1 -1 -1 -1 -1 -1 -1 -1 -1 -1]
     [-1 -1 -1 -1 -1 -1 -1 -1 -1 -1]
     [-1 -1 -1 -1 -1 -1 -1 -1 -1 -1]
     [-1 -1 -1 -1 -1 -1 -1 -1 -1 -1]
     [-1 -1 -1 -1 -1 -1 -1 -1 -1 -1]
     [-1 -1 -1 -1 -1 -1 -1 -1 -1 -1]]
```
It ignored those non-markers-connected zero-valued pixels. 
Thanks.

**Comment by emmanuelle** [comment]
Thanks @winston-li I'll see if it's possible to revert to the old behavior; maybe a warning should be triggered however when non-connected regions are turned into masked pixels of the background (as in 0.13).

**Comment by soupault** [comment]
Should be closed via #3500 .

**Comment by winston-li** [comment]
Is it fixed in scikit-image v0.14.2? The same ValueError is still raised while executing the code snippet (in 'example code to reproduce') after installed scikit-image-0.14.2...

**Comment by jni** [comment]
@winston-li no, this will be fixed in 0.15, which we hope to release within 24h. Apologies for the wait and watch this space!
